flask/overviewer/avatar.py

```diff
@@ -21,7 +21,9 @@ def get_skin_url(uuid):
     if response.status_code == 200:
         profile = json.loads(response.content.decode('utf-8', 'ignore'))
         properties = profile["properties"]
-        textures, *_ = filter(lambda obj: obj["name"] == "textures", properties)
+        textures = next(filter(lambda obj: obj["name"] == "textures", properties), None)
```

Comment:
What is `next` doing here? Can you show me the actual JSON of a player without a skin so I can test this?

Reply:
According to the documentation for [filter(function, iterable)](https://docs.python.org/3/library/functions.html#filter):
> Construct an iterator from those elements of _iterable_ for which _function_ returns true.

Therefore using [next(iterator[, default])](https://docs.python.org/3/library/functions.html#next) instead of [iterator unpacking](https://docs.python.org/3/tutorial/controlflow.html#unpacking-argument-lists) allows using a default value (`None`):
> Retrieve the next item from the _iterator_ by calling its `__next__()` method. If _default_ is given, it is returned if the iterator is exhausted, otherwise `StopIteration` is raised.

Previously textures was not set if the iterator is empty.